sdk/core/azure-core/inc/azure/core/credentials.hpp

```diff
@@ -64,13 +64,13 @@ namespace Azure { namespace Core {
   /**
    * @brief An exception that gets thrown when authentication error occurs.
    */
-  class AuthenticationException : public std::runtime_error {
+  class AuthenticationException : public std::exception {
   public:
     /**
      * @brief Construct with message string.
      *
      * @param msg Message string.
      */
-    explicit AuthenticationException(std::string const& msg) : std::runtime_error(msg) {}
+    explicit AuthenticationException(std::string const& msg) : std::exception(msg.c_str()) {}
```

Comment:
std::exception doesn't have a constructor that takes char* or string

If you want to derive rirectly from std::exception you need to implement the default method what() and save the msg inside the AuthenticationException.

So, maybe create an AzureCoreException that derives from std::exception where we store the msg and implement what().
Then we use that one instead of std::exception. 
Otherwise, if our exceptions will derive from std::exception, we will need to define/implement the same in most of them

Reply:
Thank you Victor, you are right, that constructor is not defined in standard. For now, I'll get away with overriding `what()` - I don't think we reached agreement on bas class for AzureSDK exceptions, but I am personally support the idea.